### ISSUE TITLE
Can't Regexp::Common::FIRSTKEY at /opt/mh/bin/../lib/json_server.pl line 988.

### ISSUE THREAD

**Issue by rob-vh** [hollie/misterhouse#723]
With the latest git pull (v4.2-137-g77f0f73), the List Global Variables button ends in

Can't Regexp::Common::FIRSTKEY at /opt/mh/bin/../lib/json_server.pl line 988.
 at /opt/mh/bin/mh line 31, <GEN89> line 11.
	main::__ANON__('Can\'t Regexp::Common::FIRSTKEY at /opt/mh/bin/../lib/json_se...') called at /usr/share/perl/5.18/Carp.pm line 100
	Carp::croak('Can\'t Regexp::Common::FIRSTKEY') called at /opt/mh/bin/../lib/site/Regexp/Common.pm line 125
	Regexp::Common::AUTOLOAD('Regexp::Common=ARRAY(0x462f520)') called at /opt/mh/bin/../lib/json_server.pl line 988
	main::json_walk_var('HASH(0x462fc70)', 'RE', 'ARRAY', 'HASH', 'SCALAR') called at /opt/mh/bin/../lib/json_server.pl line 924
	main::json_walk_var('*main::RE', 'RE') called at /opt/mh/bin/../lib/json_server.pl line 623
	main::json_get('GET', 'ARRAY(0x40a59b8)', 'HASH(0x40a6778)', undef) called at /opt/mh/bin/../lib/json_server.pl line 104
	main::json() called at /opt/mh/bin/../lib/http_server.pl line 534
	main::http_process_request('IO::Socket::INET=GLOB(0x40b0c40)') called at /opt/mh/bin/mh line 2565
	main::check_for_socket_data_http called at /opt/mh/bin/mh line 1682
	main::check_for_action called at /opt/mh/bin/mh line 3804
	main::monitor_commands called at /opt/mh/bin/mh line 7767

**Comment by hplato** [comment]
can you turn on debug=json and post the output before the crash?

**Comment by rob-vh** [comment]
[debug_json.txt](https://github.com/hollie/misterhouse/files/1197510/debug_json.txt)

**Comment by hplato** [comment]
What path are you using?

/ia7/#path=/vars

or

/ia7/#path=/vars_global

**Comment by hplato** [comment]
Also, the mh process can't write to your directory to update the IA7 collections file:

Warning, could not open file_write ./../data/web/collections.json: Permission denied

**Comment by hplato** [comment]
So, because the collections file isn't updating, you're using the old /vars path instead of the nicer /vars_global path. /vars tries to enumerate everything like menus, so that was cleaned up with /vars_global and it displays nicer. It seems like there are REGEX?s in your config that it's trying to parse. (I don't get the same problem so I can't diagnose).

Try adding this line 988 in json_server.pl to skip RE types:

        return if (($name eq "RE" ) and ($rtype eq "RE"));

**Comment by rob-vh** [comment]
I can see a collections.json file in my /opt/mh/data/web/ directory.  This is the file that gets refreshed by git, the file owner is root.  My services never run with root, and never update distribution materials.  Do you perhaps expect that this file is copied into /home/mh/data/web ?  Is that a manual step that the local administrator should do before starting to use ia7?
I did not add an /ia7/#path command to my mh.private.ini.  Should I have?
The link to List Global Variables in the panel is (as you predicted) http://server1.lan:8080/ia7/#path=/vars&_collection_key=0,1,29

**Comment by hplato** [comment]
The user-configurable collections.json file can exist in three places;

$Pgm_Root/data/web/
$config_parms{data_dir}/web/
$config_parms{ia7_data_dir}/

You are right, a distributed file shouldn't be changed, and really this file should be updated to 4.3, so I'll update master later.

Does /ia7/#path=/vars_global work?
Did that fix to json_server.pl work?

**Comment by rob-vh** [comment]
Manually changing the URL to vars_global works and the display looks nice.  No crash.

**Comment by hplato** [comment]
Be good to fix the crash in case others use the old method. Can you try adding that line on your setup to see if the problematic hash can be skipped? The code block should look like this:

    elsif ( $type eq 'HASH' ) {
        unless (($name eq "RE" ) and ($rtype eq "RE")) {
            foreach my $key ( sort keys %$ref ) {
                $iname = "$name\{'$key'\}";
                $iref  = \${$ref}{$key};
                $iref  = ${$ref}{$key} if ref $iref eq 'REF';
                my ( $k, $r ) = &json_walk_var( $iref, $iname, @types );
                $json_vars{$name}{$key} = $r;
            }
        }
    }

**Comment by rob-vh** [comment]
No, that doesn't cut it.  After copying jason_server.pl to my code directory, replacing the block, restarting:

```
Can't Regexp::Common::FIRSTKEY at (eval 644) line 6376.
 at /opt/mh/bin/mh line 31, <GEN42> line 11.
	main::__ANON__('Can\'t Regexp::Common::FIRSTKEY at (eval 644) line 6376.\x{a}') called at /usr/share/perl/5.18/Carp.pm line 100
	Carp::croak('Can\'t Regexp::Common::FIRSTKEY') called at /opt/mh/bin/../lib/site/Regexp/Common.pm line 125
	Regexp::Common::AUTOLOAD('Regexp::Common=ARRAY(0x44c17b8)') called at (eval 644) line 6376
	main::json_walk_var('HASH(0x42fa4f8)', 'RE', 'ARRAY', 'HASH', 'SCALAR') called at (eval 644) line 6311
	main::json_walk_var('*main::RE', 'RE') called at (eval 644) line 6010
	main::json_get('GET', 'ARRAY(0x490e620)', 'HASH(0x502dfb8)', undef) called at (eval 644) line 5491
	main::json() called at /opt/mh/bin/../lib/http_server.pl line 534
	main::http_process_request('IO::Socket::INET=GLOB(0x5106fc8)') called at /opt/mh/bin/mh line 2565
	main::check_for_socket_data_http called at /opt/mh/bin/mh line 1682
	main::check_for_action called at /opt/mh/bin/mh line 3804
	main::monitor_commands called at /opt/mh/bin/mh line 7767

```
I am not aware I added any regexps.

**Comment by hplato** [comment]
That’s an error in the new vars_global section, and the weird thing is line 644 on my json_server.pl is a comment. Is this a problem with /ia7/#path=/vars_global?

                $value =~ s/[\r\n]+$//;
                #print "db: [$key -> $value]\n";
                $json_vars_global{$key} = $value;

If so, I'm assuming $json_vars_global{$key} = $value; is causing the error. Can you uncomment that line so we can see the key and value -- those problematic variables might need to be skipped in that section as well.

And to check, does the original 

/ia7/#path=/vars still give an error?

I know turning debug=json creates a lot of text, however turning that on to see what section it's choking on is key to figuring out how to filter out the problem.

**Comment by rob-vh** [comment]
The strange line numbers are because I had to delete 3 lines from the .pl file after copying it into my private code directory, sorry for that.  Got unexplainable syntax error pointing to 3 lines of comment in the header of the.pl.  Anyway, I applied the same changes (remove # in line 644, insert an _unless_ at line 899) in the /opt/mh/lib directory this time.  debug_json.txt is the crash (still occurs).  global_json.txt is when I go to http://server1.lan:8080/ia7/#path=/vars_global&_collection_key=0,1,29
[debug_json.txt](https://github.com/hollie/misterhouse/files/1202471/debug_json.txt)
[global_json.txt](https://github.com/hollie/misterhouse/files/1202472/global_json.txt)

**Comment by hplato** [comment]
so vars_global works?

Those 'RE' types aren't being filtered out since rtype type should be REF not RE. Try changing that line to:

   print "json: DB Hash check name=[$name] rtype=[$rtype]\n";
    unless (($name eq "RE" ) and ($rtype eq "REF")) {

I still don't know why we've only seen issue on your configuration. Do you have any objects, or variables called just RE ? This check might just mask the symptom and not the root cause.

**Comment by rob-vh** [comment]
I looked better, I have 1 regexp defined in my user code:
 if ( $RE{num}{int}->matches($ventiel_state) ) { }

However, with   unless (($name eq "RE" ) and ($rtype eq "REF")) { ... }
the crash is prevented.  Output shows there is a RE type REF:

json: DB Hash check name=[cm11_objects] rtype=[REF]
json: DB Hash check name=[config_parms] rtype=[REF]
json: DB Hash check name=[config_parms_startup] rtype=[REF]
json: DB Hash check name=[Cookies] rtype=[REF]
json: DB Hash check name=[Debug] rtype=[REF]
json: DB Hash check name=[Disabled_Commands] rtype=[REF]
json: DB Hash check name=[Electric] rtype=[REF]
json: DB Hash check name=[Flags] rtype=[REF]
json: DB Hash check name=[Generic_Devices] rtype=[REF]
json: DB Hash check name=[Http] rtype=[REF]
json: DB Hash check name=[HTTP_ARGV] rtype=[REF]
json: DB Hash check name=[Included_HTML] rtype=[REF]
json: DB Hash check name=[Info] rtype=[REF]
json: DB Hash check name=[json_table] rtype=[REF]
json: DB Hash check name=[Menus] rtype=[REF]
json: DB Hash check name=[Menus{'menu_data'}] rtype=[REF]
json: DB Hash check name=[Menus{'menu_data'}{'defaults'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Android'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Android'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Android'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Android'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Android'}{'items'}[3]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Android'}{'items'}[4]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Android'}{'items'}[5]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Home_Network'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Home_Network'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[3]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[4]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[5]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[6]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[7]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[8]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[9]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[10]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[11]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[12]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[13]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[14]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[15]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[16]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[17]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[18]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[19]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[20]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[21]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[22]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[23]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[24]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[25]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[26]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'MisterHouse'}{'items'}[27]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Time'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Time'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Time'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Time'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Time'}{'items'}[3]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'Time'}{'items'}[4]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'_menu_list_states'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'mh'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'mh'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'mh'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'mh'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'mh'}{'items'}[3]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[3]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[4]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[5]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[6]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[7]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[8]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[9]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0001'}{'items'}[10]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[3]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[4]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[5]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[6]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[7]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[8]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[9]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0002'}{'items'}[10]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}{'items'}[3]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}{'items'}[4]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}{'items'}[5]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}{'items'}[6]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}{'items'}[7]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}{'items'}[8]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0003'}{'items'}[9]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0004'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0004'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0004'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0004'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0004'}{'items'}[3]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0005'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0005'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0005'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0006'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0006'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0006'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0006'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0007'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0007'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0007'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0008'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0008'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0008'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0009'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0009'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0009'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0009'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0010'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0010'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0010'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0010'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0011'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0011'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0011'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0012'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0012'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0012'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0012'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[0]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[1]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[2]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[3]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[4]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[5]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[6]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[7]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[8]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[9]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[10]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[11]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[12]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[13]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[14]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[15]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[16]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[17]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[18]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[19]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[20]] rtype=[REF]
json: DB Hash check name=[Menus{'mh'}{'states0013'}{'items'}[21]] rtype=[REF]
json: DB Hash check name=[Misc] rtype=[REF]
json: DB Hash check name=[Moon] rtype=[REF]
json: DB Hash check name=[Password_Allow] rtype=[REF]
json: DB Hash check name=[Persistent] rtype=[REF]
json: DB Hash check name=[RE] rtype=[REF]
json: DB Hash check name=[Run_Members] rtype=[REF]
json: DB Hash check name=[Save] rtype=[REF]
json: DB Hash check name=[Serial_Ports] rtype=[REF]
json: DB Hash check name=[socket_fork_data] rtype=[REF]
json: DB Hash check name=[Socket_Ports] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'http'}] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'localhost:8080'}] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'server_advantech_LBK'}] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'server_advantech_ketel'}] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'server_android'}] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'server_mhsend'}] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'server_telnet'}] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'xap_listen_core'}] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'xap_send'}] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'xpl_listen'}] rtype=[REF]
json: DB Hash check name=[Socket_Ports{'xpl_send'}] rtype=[REF]
json: DB Hash check name=[Sounds] rtype=[REF]
json: DB Hash check name=[ti103_objects] rtype=[REF]
json: DB Hash check name=[Tk_objects] rtype=[REF]
json: DB Hash check name=[Tk_results] rtype=[REF]
json: DB Hash check name=[triggers] rtype=[REF]
json: DB Hash check name=[triggers{'Afzuig normaal'}] rtype=[REF]
json: DB Hash check name=[triggers{'Afzuig normaal 2'}] rtype=[REF]
json: DB Hash check name=[triggers{'Afzuig normaal 3'}] rtype=[REF]
json: DB Hash check name=[triggers{'BG Huiskamer Comf Off'}] rtype=[REF]
json: DB Hash check name=[triggers{'BG Keuken Comf'}] rtype=[REF]
json: DB Hash check name=[triggers{'BG Keuken Comf Off'}] rtype=[REF]
json: DB Hash check name=[triggers{'BG Werkkamer Comf'}] rtype=[REF]
json: DB Hash check name=[triggers{'BG Werkkamer Comf Off'}] rtype=[REF]
json: DB Hash check name=[triggers{'Boiler pomp uit'}] rtype=[REF]
json: DB Hash check name=[triggers{'Keuken Comfort Off'}] rtype=[REF]
json: DB Hash check name=[triggers{'Koeling aan'}] rtype=[REF]
json: DB Hash check name=[triggers{'Koeling uit'}] rtype=[REF]
json: DB Hash check name=[triggers{'Rondleiding aan'}] rtype=[REF]
json: DB Hash check name=[triggers{'Rondleiding uit'}] rtype=[REF]
json: DB Hash check name=[triggers{'Update current weather via WeatherBug'}] rtype=[REF]
json: DB Hash check name=[triggers{'Update weather forecast via WeatherBug'}] rtype=[REF]
json: DB Hash check name=[triggers{'Ventilatie Dag'}] rtype=[REF]
json: DB Hash check name=[triggers{'Ventilatie Nacht'}] rtype=[REF]
json: DB Hash check name=[triggers{'Verw BG Dag'}] rtype=[REF]
json: DB Hash check name=[triggers{'Verw BG Nacht'}] rtype=[REF]
json: DB Hash check name=[triggers{'Verw BG Nacht 2'}] rtype=[REF]
json: DB Hash check name=[triggers{'Verw V Dag'}] rtype=[REF]
json: DB Hash check name=[triggers{'Verw V Nacht'}] rtype=[REF]
json: DB Hash check name=[triggers{'get MH version'}] rtype=[REF]
json: DB Hash check name=[triggers{'update the documentation'}] rtype=[REF]
json: DB Hash check name=[Weather] rtype=[REF]

**Comment by hplato** [comment]
The proper way to solve this is to wrap json_walk_var in an eval loop to prevent against any other data that causes problem with the regex module. An error should still be thrown for $RE, but it shouldn't crash MH. 

I updates json_server.pl on a gist. Can you try this file and see if it crashes MH with you try #path=/vars and #path=/vars_global

https://gist.github.com/hplato/a1cd4bb72ce1b44c2de4c9f21ac1fc76

**Comment by rob-vh** [comment]
This new version of json_server.pl works, when I go to http://server1.lan:8080/ia7/#path=/vars&_collection_key=0,1,29 I get
```
08/06/17 10:05:33 AM Json_Server.pl: WARNING: JSON variable parsing: RE failed to process
08/06/17 10:05:33 AM Json_Server.pl: WARNING: JSON variable parsing: RE failed to process
08/06/17 10:05:34 AM Json_Server.pl: WARNING: JSON variable parsing: RE failed to process
08/06/17 10:05:35 AM Json_Server.pl: WARNING: JSON variable parsing: RE failed to process
08/06/17 10:05:36 AM Json_Server.pl: WARNING: JSON variable parsing: RE failed to process
08/06/17 10:05:37 AM Json_Server.pl: WARNING: JSON variable parsing: RE failed to process
08/06/17 10:05:37 AM Json_Server.pl: WARNING: JSON variable parsing: RE failed to process
08/06/17 10:05:38 AM Json_Server.pl: WARNING: JSON variable parsing: RE failed to process
08/06/17 10:05:39 AM Json_Server.pl: WARNING: JSON variable parsing: RE failed to process
08/06/17 10:05:40 AM Json_Server.pl: WARNING: JSON variable parsing: RE failed to process
```
and no crash.

**Comment by hplato** [comment]
Fixed in #724
